userLoggedInMainPage.php

```diff
@@ -1,24 +1,22 @@
-<?php session_start();
+<?php  
     include('config/init.php');
     echoTrackerHeader("Login");
 
  
 //$accountId=getUser($_REQUEST['accountId']); //save number in url as variable
-/* $accountId=$_SESSION['accountId'];
+/* $accountId=$_SESSION['accountId']; */
+//$user = getUser($accountId); 
 
 
 
-echo "hello user".$accountId." More coming soon."; */
+echo " 
 
-
-
-
-
-echo "hello user ".$_SESSION['accountId']['email'].". More coming soon.";
+hello user ".$_SESSION['accountId']['email'].". More coming soon.";
```

Comment:
It looks like you're storing the entire user record in the session. This technically works, but it's not ideal for a few reasons:

- There's a security concern in putting the user's password in the session. Even if it's hashed, the more places you store data, the more likely it is that you mess up and leak it
- It's possible that you end up having a ton of data about a user and the session storage isn't meant to store a lot
- (most important) - you now have two sources of true. Like, imagine a user logs in and changes their email. The session will still have their old email, and you might not know which one to trust (the session or the database)

Instead, you should only store the `accountId` in the session. At the top of any page where you need user info, call `getUser($_SESSION['accountId']);`. That will give you the info you want to echo here without needing to store it directly in the session.